pagopa/pn-bff
# PN-13067 - Migrate from getLegalFact method to getLegalFactById

## Short description
Migrate from the `getLegalFact` method to `getLegalFactById`, as `getLegalFact` is now deprecated.

## List of changes proposed in this pull request
- Removed `legalFactCategory` definition from the OpenAPI specification
- Removed `legalFactCategory` from all associated methods
- Refactored tests to align with the updated method

## How to test
1. Compile the code.
2. Use Postman to send requests to:
  - http://localhost:8091/bff/v1/notifications/received/TKZX-DLGT-KPLW-202410-Z-1/documents/LEGAL_FACT?documentId=PN_LEGAL_FACTS-811d13d6238e43a09a3d529222fc76fc.pdf
  (`x-pagopa-pn-uid: dfa2d7f4-7c1e-4fef-be30-60ead786e70f`, `x-pagopa-pn-cx-type: PF`, `x-pagopa-pn-cx-id: PF-dfa2d7f4-7c1e-4fef-be30-60ead786e70f`, and empty `x-pagopa-pn-cx-groups`)
  - http://localhost:8091/bff/v1/notifications/sent/HXJX-ZWLG-DJAJ-202410-A-1/documents/LEGAL_FACT?documentId=PN_LEGAL_FACTS-8ccbd6086a2a400a82f8379d4e6458de.pdf
  (`x-pagopa-pn-uid: e9e4a9c7-9586-4b92-a7dd-ee1a0e77d398`, `x-pagopa-pn-cx-type: PA`, `x-pagopa-pn-cx-id: 5b994d4a-0fa8-47ac-9c7b-354f1d44a1ce`, and empty `x-pagopa-pn-cx-groups`)
3. Confirm that both requests return a 200 status and include a populated response body.
